common/base64.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * `b64.h' - b64
+ *
+ * copyright (c) 2014 joseph werle
+ */
+
+#ifndef B64_H
+#define B64_H 1
+
+/**
+ *  Memory allocation functions to use. You can define b64_malloc and
+ * b64_realloc to custom functions if you want.
+ */
+
+#ifndef b64_malloc
```

Comment:
We should be using only `tal` and related functions.  So things like `b32_encode` should accept a `const tal_t *ctx` argument that will own the returned encoded string.

Reply:
wip